test/e2e/apps/statefulset.go

```diff
@@ -882,6 +889,89 @@ var _ = SIGDescribe("StatefulSet", func() {
 			appTester.run()
 		})
 	})
+
+	framework.KubeDescribe("V1beta2 StatefulSet Scale Subresource [Feature:StatefulSet]", func() {
+		ssName := "ss"
+		labels := map[string]string{
+			"foo": "bar",
+			"baz": "blah",
+		}
+		headlessSvcName := "test"
+		var statefulPodMounts, podMounts []v1.VolumeMount
+		var ss *appsV1beta2.StatefulSet
+
+		BeforeEach(func() {
+			statefulPodMounts = []v1.VolumeMount{{Name: "datadir", MountPath: "/data/"}}
+			podMounts = []v1.VolumeMount{{Name: "home", MountPath: "/home"}}
+			ss = framework.NewStatefulSetV1beta2(ssName, ns, headlessSvcName, 2, statefulPodMounts, podMounts, labels)
+
+			By("Creating service " + headlessSvcName + " in namespace " + ns)
+			headlessService := framework.CreateServiceSpec(headlessSvcName, "", true, labels)
+			_, err := c.Core().Services(ns).Create(headlessService)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			if CurrentGinkgoTestDescription().Failed {
+				framework.DumpDebugInfo(c, ns)
+			}
+			framework.Logf("Deleting all statefulset in ns %v", ns)
+			framework.DeleteAllStatefulSets(c, ns)
+		})
+
+		It("should be modifiable", func() {
+			By("Creating statefulset " + ssName + " in namespace " + ns)
+			*(ss.Spec.Replicas) = 1
+			sst := framework.NewStatefulSetTester(c)
+			sst.PauseNewPodsV1beta2(ss)
+
+			_, err := c.AppsV1beta2().StatefulSets(ns).Create(ss)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("starting proxy server")
+			port, cmd, err := startProxyServer()
+			if cmd != nil {
+				defer framework.TryKill(cmd)
+			}
+			if err != nil {
+				framework.Failf("Failed to start proxy server: %v", err)
+			}
+
+			ssPath := fmt.Sprintf("http://localhost:%d/apis/apps/v1beta2/namespaces/default/statefulsets/%s/scale", port, ssName)
+			curlCmd := "curl"
+			args := []string{"-X", "GET", ssPath}
+			var body []byte
+			var scaleSubresourceSpec map[string]interface{}
+
+			By("curling proxy to obtain Spec.Replicas info")
+			body, err = exec.Command(curlCmd, args...).Output()
+			if err != nil {
+				framework.Failf("Failed to curl proxy to obtain Spec.Replicas info: %v", err)
+			}
+			err = json.Unmarshal([]byte(body[:]), &scaleSubresourceSpec)
+			if err != nil {
+				framework.Failf("Failed to unmarshal JSON blob: %v", err)
+			}
+			spec := scaleSubresourceSpec["spec"].(map[string]interface{})
+			Expect((spec["replicas"]).(float64)).To(Equal(1.0))
+
+			By("changing Spec.Replicas from 1 to 9 via PUT (update)")
+			spec["replicas"] = 9.0
+			body, err = json.Marshal(scaleSubresourceSpec)
+			Expect(err).NotTo(HaveOccurred())
+			putArgs := []string{"-X", "PUT", "-d", string(body), ssPath, "-H", "Content-type: application/json"}
+			_, err = exec.Command(curlCmd, putArgs...).Output()
+			Expect(err).NotTo(HaveOccurred())
+
+			By("curling proxy again to verify change to Spec.Replicas")
```

Comment:
@enisoc i carried out a manual testing and confirmed that the change reflected to both `StatefulSet.Spec` and `Scale.Spec`. i will look more into `RESTClient` for e2e testing.